platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/IsolatedCodecs.kt

```diff
@@ -18,10 +18,10 @@ package org.gradle.configurationcache.serialization.codecs
 
 import com.google.common.collect.ImmutableList
 import com.google.common.collect.ImmutableSet
-import org.gradle.configurationcache.serialization.Codec
-import org.gradle.configurationcache.serialization.ReadContext
-import org.gradle.configurationcache.serialization.WriteContext
-import org.gradle.configurationcache.serialization.readNonNull
+import org.gradle.internal.serialize.isolate.Codec
+import org.gradle.internal.serialize.isolate.ReadContext
+import org.gradle.internal.serialize.isolate.WriteContext
+import org.gradle.internal.serialize.isolate.readNonNull
 import org.gradle.internal.hash.HashCode
 import org.gradle.internal.isolation.Isolatable
```

Comment:
I noticed we already have the concept of "isolatable" objects, that seems completely unrelated (implementation-wise, even if shares goals) to what we now want to call "isolates". 

We are now promoting this concept beyond a mere module-private concept and as a codebase-wide concept. I think that may generate confusion.
 
I liked "object serialization" or " graph serialization" better. If we really want to introduce or promote a new concept, we could use some additional analysis and discussion, and focus now on just addressing the lack of an "internal" segment in package names.

Reply:
I understand the point that _isolatable_ and _isolate_ sound like similar and/or related concepts, however I don't immediately agree the concept of `isolate` shouldn't be promoted here since it's one of the main features (if not the main feature) required and used by the Configuration Cache (the ability to isolate tasks so they can be executed in parallel), and the lifecycle callbacks (although not as prominently and explicitly).

Let's have that conversation and then apply a new rename if needed. So far we are the only users of this library.

Reply:
On a second thought, I think there's merit to what you said. We have two serialization libraries, one for basic data types and this one. When deciding which one to use, the 1st question is likely "do I need to serialize graphs or not?" rather than "do I need to isolate portions of the graph?". I'll rename it to `graph-serialization` and the package to `o.g.i.serialize.graph`.